lib/scan_beacon/bluez_advertiser.rb

```diff
@@ -1,54 +1,34 @@
 module ScanBeacon
-  class BlueZAdvertiser
+  class BlueZAdvertiser < GenericIndividualAdvertiser
  
-    attr_accessor :beacon, :parser, :ad, :addr
+    attr_reader :addr
 
     def initialize(opts = {})
       @device_id = opts[:device_id] || BlueZ.devices.map {|d| d[:device_id]}[0]
       raise "No available devices" if @device_id.nil?
       BlueZ.device_up @device_id
-      @addr = @initial_addr = BlueZ.devices.find {|d| d[:device_id] == @device_id}[:addr]
-      self.beacon = opts[:beacon]
-      self.parser = opts[:parser]
-      self.parser ||= BeaconParser.default_parsers.find {|parser| parser.beacon_type == beacon.beacon_types.first}
+      addr = @initial_addr = BlueZ.devices.find {|d| d[:device_id] == @device_id}[:addr]
+      super(opts)
     end
 
-    def beacon=(value)
-      @beacon = value
-      update_ad
-    end
-
-    def parser=(value)
-      @parser = value
-      update_ad
-    end
-
-    def ad=(value)
-      @ad = value
+    def start(with_rotation = false)
+      @addr = random_addr if with_rotation
       BlueZ.set_advertisement_bytes @device_id, @ad
-    end
-
-    def start
-      BlueZ.start_advertising @device_id, nil
-    end
-
-    def start_with_random_addr
-      addr = random_addr
+      # You must call start advertising any time you change the advertisement bytes
+      # otherwise they won't take
       BlueZ.start_advertising @device_id, addr
```

Comment:
Passing an `addr` into `start_advertising` also tells the hardware that we are supplying a random address.  I'm not sure we want to do that unless it really is a random address.  I think it needs to be `nil` if we want the default address.